Neos.ContentRepository.NodeMigration/src/NodeMigrationServiceFactory.php

```diff
@@ -7,50 +7,79 @@
 use Neos\ContentRepository\NodeMigration\Transformation\RemovePropertyTransformationFactory;
 use Neos\ContentRepository\NodeMigration\Transformation\RenameNodeAggregateTransformationFactory;
 use Neos\ContentRepository\NodeMigration\Transformation\RenamePropertyTransformationFactory;
 use Neos\ContentRepository\NodeMigration\Transformation\StripTagsOnPropertyTransformationFactory;
+use Neos\ContentRepository\NodeMigration\Transformation\TransformationFactoryInterface;
 use Neos\ContentRepository\NodeMigration\Transformation\TransformationsFactory;
 use Neos\ContentRepository\NodeMigration\Transformation\UpdateRootNodeAggregateDimensionsTransformationFactory;
 
 /**
  * @implements ContentRepositoryServiceFactoryInterface<NodeMigrationService>
  */
-class NodeMigrationServiceFactory implements ContentRepositoryServiceFactoryInterface
+final readonly class NodeMigrationServiceFactory implements ContentRepositoryServiceFactoryInterface
 {
+    /**
+     * @param array<string,class-string<FilterFactoryInterface>> $filterFactories
+     * @param array<string,class-string<TransformationFactoryInterface|PropertyConverterAwareTransformationFactoryInterface>> $transformationFactories
+     */
+    public function __construct(
+        private array $filterFactories,
+        private array $transformationFactories
+    ) {
+    }
+
+    public static function createDefault(): self
+    {
+        return new self(
+            filterFactories: [
+                'DimensionSpacePoints' => DimensionSpacePointsFilterFactory::class,
+                'NodeName' => NodeNameFilterFactory::class,
+                'NodeType' => NodeTypeFilterFactory::class,
+                'PropertyNotEmpty' => PropertyNotEmptyFilterFactory::class,
+                'PropertyValue' => PropertyValueFilterFactory::class,
+            ],
+            transformationFactories: [
+                'AddDimensionShineThrough' => AddDimensionShineThroughTransformationFactory::class,
+                'AddNewProperty' => AddNewPropertyConverterAwareTransformationFactory::class,
+                'ChangeNodeType' => ChangeNodeTypeTransformationFactory::class,
+                'ChangePropertyValue' => ChangePropertyValueConverterAwareTransformationFactory::class,
+                'MoveDimensionSpacePoint' => MoveDimensionSpacePointTransformationFactory::class,
+                'RemoveNode' => RemoveNodeTransformationFactory::class,
+                'RemoveProperty' => RemovePropertyTransformationFactory::class,
+                'RenameNodeAggregate' => RenameNodeAggregateTransformationFactory::class,
+                'RenameProperty' => RenamePropertyTransformationFactory::class,
+                'StripTagsOnProperty' => StripTagsOnPropertyTransformationFactory::class,
+                'UpdateRootNodeAggregateDimensions' => UpdateRootNodeAggregateDimensionsTransformationFactory::class,
+            ]
+        );
+    }
```

Comment:
Why is this needed if we already have that configured in the settings?

Reply:
i wasnt sure how independent or not the behavioural test should be from the configuration (currently they use this factory) and seeing the migrations package could also be used standalone i assume it makes sense to provide a default?

Reply:
IMHO I'd remove this. If the tests do need the configuration, we should provide it in the tests.

Reply:
This could be important and necessary if people chose to overwrite transformations/filters via packages, so that suddenly certain tests fail if such a package is installed...